test/check-application

```diff
@@ -162,6 +162,111 @@ class TestApplication(testlib.MachineCase):
         b.wait_in_text("#certificate-0-cert-storage", "FILE")
         b.wait_in_text("#certificate-0-cert-location", cert_path)
 
+    def testRequestCert(self):
+        b = self.browser
+        m = self.machine
+
+        self.login_and_go("/certificates")
+        # verify expected heading
+        b.wait_in_text(".ct-table-header h3", "Certificates")
+
+        class RequestCertDialog(object):
+            def __init__(
+                self, test_obj, ca="SelfSign", storage_type="NSSDB", nickname="",
+                key_path="", cert_path=""
+            ):
+                self.test_obj = test_obj
+                self.ca = ca
+                self.storage_type = storage_type
+                self.nickname = nickname
+                self.cert_path = cert_path
+                self.key_path = key_path
+
+            def execute(self):
+                self.open()
+                self.fill()
+                self.create()
+                self.verify_frontend()
+                self.verify_backend()
+                self.cleanup()
+
+            def open(self):
+                b.click("#request-certificate-action")
+                b.wait_in_text(".modal-dialog .modal-header .modal-title", "Request Certificate")
+
+            def fill(self):
+                b.select_from_dropdown("#ca", self.ca)
+                b.click(".pf-c-radio__input + label:contains({0})".format(self.storage_type))
+
+                if self.storage_type == "NSSDB":
+                    b.set_input_text("#nickname", self.nickname)
+                else:
+                    b.set_input_text("input[placeholder='Path to store the certificate']", self.cert_path)
+                    b.set_input_text("input[placeholder='Path to store the generated key or to an existing key']", self.key_path)
+
+            def cancel(self):
+                b.click(".modal-footer button:contains(Cancel)")
+                b.wait_not_present(".modal-dialog")
+
+            def create(self):
+                b.click(".modal-footer button:contains(Request)")
+                b.wait_not_present(".modal-dialog")
+                time.sleep(4) # Certificate enrollment takes some time
```

Comment:
This is not ideal. There should be other way how to wait for stuff.
If it does not propagate anyhow in the UI, then there is surely something we can wait for on the commandline.

That being said, I don't think we need to block on that, would be just nice to fix eventually.